File: test-server.py
#!/usr/bin/env python3
"""
Simple HTTP server for testing the Fixers in Greece website locally
"""

import http.server
import socketserver
import os
import webbrowser
from urllib.parse import unquote

PORT = 8000
HOST = "localhost"

class MyHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def end_headers(self):
        # Add headers to prevent caching during development
        self.send_header('Cache-Control', 'no-store, no-cache, must-revalidate')
        self.send_header('Expires', '0')
        super().end_headers()
    
    def log_message(self, format, *args):
        # Custom logging format
        print(f"[{self.log_date_time_string()}] {format % args}")
    
    def do_GET(self):
        # Decode URL to handle special characters
        self.path = unquote(self.path)
        
        # Handle directory requests by serving index.html
        if self.path.endswith('/'):
            self.path += 'index.html'
        
        return super().do_GET()

def start_server():
    """Start the test server"""
    # Change to the project directory
    web_dir = os.path.dirname(os.path.abspath(__file__))
    os.chdir(web_dir)
    
    Handler = MyHTTPRequestHandler
    
    print(f"\n🚀 Starting Fixers in Greece Test Server...")
    print(f"📁 Serving files from: {web_dir}")
    print(f"🌐 Server URL: http://{HOST}:{PORT}")
    print(f"\n✨ Opening browser automatically...")
    print(f"📝 Press Ctrl+C to stop the server\n")
    
    # Open browser automatically
    webbrowser.open(f'http://{HOST}:{PORT}')
    
    with socketserver.TCPServer(("", PORT), Handler) as httpd:
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print("\n\n🛑 Server stopped by user")
            return

if __name__ == "__main__":
    start_server()

File: test-server.js
const http = require('http');
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

const PORT = 8000;
const HOST = 'localhost';

// MIME types for different file extensions
const mimeTypes = {
    '.html': 'text/html',
    '.css': 'text/css',
    '.js': 'text/javascript',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.jpeg': 'image/jpeg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.ico': 'image/x-icon',
    '.webp': 'image/webp',
    '.mp4': 'video/mp4',
    '.webm': 'video/webm',
    '.woff': 'font/woff',
    '.woff2': 'font/woff2',
    '.ttf': 'font/ttf',
    '.otf': 'font/otf'
};

const server = http.createServer((req, res) => {
    // Decode URL
    let filePath = decodeURIComponent(req.url);
    
    // Default to index.html for directory requests
    if (filePath.endsWith('/')) {
        filePath += 'index.html';
    }
    
    // Remove query strings
    filePath = filePath.split('?')[0];
    
    // Security: prevent directory traversal
    filePath = path.join(__dirname, filePath);
    
    // Get file extension
    const extname = path.extname(filePath).toLowerCase();
    const contentType = mimeTypes[extname] || 'application/octet-stream';
    
    // Read and serve the file
    fs.readFile(filePath, (err, content) => {
        if (err) {
            if (err.code === 'ENOENT') {
                // File not found
                res.writeHead(404, { 'Content-Type': 'text/html' });
                res.end('<h1>404 - File Not Found</h1>', 'utf-8');
            } else {
                // Server error
                res.writeHead(500);
                res.end(`Server Error: ${err.code}`, 'utf-8');
            }
        } else {
            // Success
            res.writeHead(200, { 
                'Content-Type': contentType,
                'Cache-Control': 'no-store, no-cache, must-revalidate',
                'Expires': '0'
            });
            res.end(content, 'utf-8');
        }
    });
    
    // Log the request
    console.log(`[${new Date().toLocaleString()}] ${req.method} ${req.url}`);
});

server.listen(PORT, HOST, () => {
    console.log('\n🚀 Starting Fixers in Greece Test Server...');
    console.log(`📁 Serving files from: ${__dirname}`);
    console.log(`🌐 Server URL: http://${HOST}:${PORT}`);
    console.log('\n✨ Opening browser automatically...');
    console.log('📝 Press Ctrl+C to stop the server\n');
    
    // Open browser
    const url = `http://${HOST}:${PORT}`;
    const start = process.platform === 'darwin' ? 'open' : process.platform === 'win32' ? 'start' : 'xdg-open';
    exec(`${start} ${url}`);
});

// Handle server shutdown
process.on('SIGINT', () => {
    console.log('\n\n🛑 Server stopped by user');
    process.exit(0);
});